packages/@lwc/engine-core/src/framework/utils.ts

```diff
@@ -104,3 +104,13 @@ export function shouldBeFormAssociated(Ctor: LightningElementConstructor) {
 
     return ctorFormAssociated && apiFeatureEnabled;
 }
+
+// check if a property is in an object, and if the object throws an error merely because we are
+// checking if the property exists, return false
+export function safeHasProp(obj: any, prop: string) {
```

Comment:
```suggestion
export function safeHasProp<T extends object, K extends PropertyKey>(obj: T, prop: K): obj is Record<K, unknown> {
```

Reply:
I tried to do this but failed. `Record` was the thing I was missing!

Reply:
TypeScript is still angry

![Screenshot 2025-01-08 at 9 54 34 AM](https://github.com/user-attachments/assets/1cef1bb4-2615-4b05-adb5-6801948fecca)

Reply:
5edfa007f